build/tasks/node_smoke_tests.js

```diff
@@ -17,33 +18,116 @@ if ( !verifyNodeVersion() ) {
 // important so that the tests & the main process don't interfere with
 // each other, e.g. so that they don't share the `require` cache.
 
-async function runTests( sourceType, module ) {
-	if ( !allowedModules.includes( sourceType ) ) {
-		throw new Error(
-			`Usage: \`node_smoke_tests [${allowedModules.join( "|" )}]:JQUERY\``
-		);
+async function runTests( { libraryType, sourceType, module } ) {
+	if ( !allowedLibraryTypes.includes( libraryType ) ||
+			!allowedSourceTypes.includes( sourceType ) ) {
+		throw new Error( `Incorrect libraryType or sourceType value; passed: ${
+			libraryType
+		} ${ sourceType } "${ module }"` );
 	}
-	const dir = `./test/node_smoke_tests/${sourceType}`;
+	const dir = `./test/node_smoke_tests/${ sourceType }/${ libraryType }`;
 	const files = await fs.promises.readdir( dir, { withFileTypes: true } );
 	const testFiles = files.filter( ( testFilePath ) => testFilePath.isFile() );
+
+	if ( !testFiles.length ) {
+		throw new Error( `No test files found for ${
+			libraryType
+		} ${ sourceType } "${ module }"` );
+	}
+
 	await Promise.all(
 		testFiles.map( ( testFile ) =>
-			exec( `node "${dir}/${testFile.name}" "${module}"` )
```

Comment:
We should have spaces inside of interpolations in template literals as it's consistent with the jQuery code style. We should update our ESLint config to account for that.

Reply:
TBF, the jQuery eslint plugin doesn't specify this. And, while our style guide does designate spaces inside curly braces, template literals could be considered separate, and it seems eslint agrees.

Reply:
I know it does not specify this but since the jQuery style guide spirit is basically "spaces everywhere", it makes sense to cover template literals with this rule as well for consistency.

It doesn't touch on this now because for a long time the style guide was only catering to ES5.

Reply:
There are many spacing rules in ESLint and yet we don't describe each of them separately as they all form a bigger style guide for us (some rules we've had to enable: `space-before-blocks`, `space-in-parens`, `space-infix-ops`, `space-before-function-paren`, `space-unary-ops`). ESLint rules are just often quite granular.

We've actually gravitated towards consistent spacing within any kind of brackets over the years (old jQuery didn't space out the ending `});`, for example). This makes it easy to remember the rules - always put a space after `(`, `{`, `[` and before `}`, `)`, `]`, etc. - regardless of what they're used for.

Also, while the jQuery style results overall in longer lines, it has its advantages - spaces in all kinds of brackets make it easier to distinguish tokens when parsing the code with our own eyes. For the same reason, I prefer to have spaces in template literal interpolations; it makes it easier to read them for me. Especially when I'm used to this rich spacing in jQuery projects, having this one being a sole case of any opening paren not being followed by a space makes it look inconsistent to me - and it doesn't really matter that in ESLint this is a separate rule.

Reply:
Yes, I'd expect the change to be done in the plugin.